src/Npgsql/NpgsqlConnector.cs

```diff
@@ -461,6 +461,10 @@ string GetUsername()
             if (username?.Length > 0)
                 return username;
 
+            username = PostgresVariables.User;
```

Comment:
Any reason not to coalesce this above like the other variables? It looks like you're following the preexisting code, but if we can align the pattern, I would opt to do it.

Reply:
Yeah, this is a new feature which shouldn't alter unrelated code. Speaking about coalesce, password detection works differently and probably should be unified with this one.

@roji Do you remember why you did it this way? What's a reason to check a string length too?

Reply:
Yes, that's the question.

Reply:
Not sure. A zero-length password *may* be a valid password (this would require checking), but I'm pretty sure that a zero-length username isn't a valid username.

Reply:
From [the documentation](https://www.postgresql.org/docs/current/sql-createrole.html):

>Specifying an empty string will also set the password to null, but that was not the case before PostgreSQL version 10. In earlier versions, an empty string could be used, or not, depending on the authentication method and the exact version, and libpq would refuse to use it in any case. To avoid the ambiguity, specifying an empty string should be avoided.

Let's avoid empty stings too.

Reply:
Makes sense.

Reply:
Tracked by #2696.